patches/0002-Add-crypto-backend-foundation.patch

```diff
@@ -2182,23 +2182,22 @@ index 76fff6974e7403..218d40171d2567 100644
  	"crypto/internal/fips140/mlkem"
  	"crypto/internal/fips140/tls13"
  	"crypto/internal/hpke"
-@@ -477,6 +478,15 @@ func cloneHash(in hash.Hash, h crypto.Hash) hash.Hash {
+@@ -477,6 +478,14 @@ func cloneHash(in hash.Hash, h crypto.Hash) hash.Hash {
  	}
  	marshaler, ok := in.(binaryMarshaler)
  	if !ok {
 +		if boring.Enabled {
 +			// CNG and OpenSSL with SymCrypt hash functions do not implement the
 +			// encoding.BinaryMarshaler interface, but they do implement the Clone method.
-+			if cloner, ok := in.(interface{ Clone() (hash.Hash, error) }); ok {
-+				if out, err := cloner.Clone(); err == nil {
-+					return out
-+				}
++			out := in.(interface{ Clone() hash.Hash }).Clone()
```

Comment:
Would it be a good idea to keep `ok` and panic more descriptively if this fails?

Reply:
In fact I'm not 100% sure that only non-boring hash objects can get to this point, that's why the previous behavior was falling back to `MarshalBinary` if the object didn't implement the `Clone` method. I would keep the same behavior for now.